File: lib/commands/builtin/back.ts
import type { CommandSpec } from "../types"

export const backCommand: CommandSpec = {
  name: "back",
  aliases: ["b"],
  summary: "navigate the tab history backward",
  usage: ["back [tabId]", "b [tabId]"],
  man: [
    "NAME",
    "  back, b - navigate the tab history backward",
    "",
    "SYNOPSIS",
    "  back [tabId]",
    "",
    "DESCRIPTION",
    "  If tabId is omitted, uses the active tab in the last focused window",
    "  (with fallbacks; see focus)."
  ],
  async execute(ctx, args) {
    const tab = await ctx.resolveTabArg(args[1])
    if (!tab?.id) {
      return ["no target tab (set focus window or pass tabId)"]
    }
    await chrome.tabs.goBack(tab.id)
    return [`goBack tab ${tab.id}`]
  }
}

File: wasm/bmxt-core/src/cmd/back.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "back",
    aliases: &["b"],
    usage_primary: "back [tabId]",
    man: &[
        "NAME",
        "  back, b - navigate the tab history backward",
        "",
        "SYNOPSIS",
        "  back [tabId]",
        "",
        "DESCRIPTION",
        "  If tabId is omitted, uses the active tab in the last focused window",
        "  (with fallbacks; see focus).",
    ],
};

pub fn run(args: &[String]) -> DispatchJson {
    let tab_arg = args.get(1).cloned();
    DispatchJson::effects(vec![Effect::GoBack { tab_id_arg: tab_arg }])
}
